pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -144,6 +144,48 @@ type CanaryStrategy struct {
 	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
 }
 
+// RolloutCanaryExperimentStep defines a template that is used to create a experiment for a step
+type RolloutCanaryExperimentStep struct {
+	// Indicates if the rollout should wait for the experiment to finish
+	Duration int32 `json:"duration"`
+	// Templates what templates that should be added to the experiment. Should be non-nil
+	Templates []RolloutExperimentTemplate `json:"templates"`
+	//ProgressingDeadlineSeconds Is it ncessary?
+}
+
+// RolloutExperimentTemplate defines the template used to create experiments for the Rollout's experiment canary step
+type RolloutExperimentTemplate struct {
+	// Name description of template that passed to the template
+	Name string `json:"name"`
+	// Type indicates where the rollout should get the RS template from
+	SpecRef ReplicaSetSpecRef `json:"specRef"`
+	// Replicas replica count for the template
+	Replicas int32 `json:"replicas"`
+	// AdditionalSelectors additional selectors to use for the RS created from the template
+	// +optional
+	Metadata RolloutExperimentAdditionalMetadata `json:"metadata,omitempty"`
+}
+
+//RolloutExperimentAdditionalMetadata extra labels to add to the template
+type RolloutExperimentAdditionalMetadata struct {
```

Comment:
How about: `PodTemplateMetadata`. I would make clear that this metadata will go into the labels/annotations of the actual pod

Reply:
done